pdc/apps/package/serializers.py

```diff
@@ -221,3 +222,60 @@ class BuildImageRTTTestsSerializer(StrictSerializerMixin, serializers.ModelSeria
     class Meta:
         model = models.BuildImage
         fields = ('id', 'build_nvr', 'format', 'test_result')
+
+
+class RepositoryField(serializers.SlugRelatedField):
+
+    def __init__(self, **kwargs):
+        super(RepositoryField, self).__init__(slug_field='id', queryset=Repo.objects.all(), **kwargs)
+
+    def to_representation(self, value):
+        export_data = value.export()
+        export_data["id"] = value.id
+        export_data["pdc_endpoint"] = value.content_format.pdc_endpoint
+        return export_data
+
+
+class ReleasedFilesSerializer(StrictSerializerMixin, serializers.ModelSerializer):
+    file_primary_key = serializers.IntegerField(allow_null=False)
+    repo = RepositoryField()
+    released_date = serializers.DateField(required=False)
+    release_date = serializers.DateField()
+    created_at = serializers.DateTimeField(required=False, read_only=True)
+    updated_at = serializers.DateTimeField(required=False, read_only=True)
+    zero_day_release = serializers.BooleanField(required=False, default=False)
+    obsolete = serializers.BooleanField(required=False, default=False)
+
+    class Meta:
+        model = models.ReleasedFiles
+        fields = ('id', 'file_primary_key', 'repo', 'released_date',
+                  'release_date', 'created_at', 'updated_at',
+                  'zero_day_release', 'obsolete')
+
+    def validate(self, data):
+        if "repo" in data:
+            repo_format = data["repo"].content_format
+            repo_name = data["repo"].name
+            if str(repo_format) != "rpm":
+                raise serializers.ValidationError(
+                    {'detail': 'Currently we '
+                               'just support rpm type of repo, the type of %s is %s ' % (repo_name, repo_format)})
+
+        # if there are other types support, should add codes check if the primary key in right table
+
+        return data
+
+    def to_representation(self, value):
+        rep_data = super(ReleasedFilesSerializer, self).to_representation(value)
+        if "repo" in rep_data and "pdc_endpoint" in rep_data["repo"]:
+            rep_data["pdc_endpoint"] = rep_data["repo"]["pdc_endpoint"]
+            rep_data["repo"].pop("pdc_endpoint")
```

Comment:
Looks like it would be much easier to just set `rep_data["pdc_endpoint"] = "rpm"` below and remove the `pdc_endpoint` from data model.

Reply:
This is a hard code, maybe one day we change that, we don't want to change the codes.
I am thinking that as there are multi types here, we should do an automatically way to do this, but not just add the if/else.

Reply:
Why would we want to change that? That would break API. Currently, we need to change the code anyway to support new content type.

Reply:
I just think anything could happen :), also when we design a new endpoint with new type, we can't make sure if we never change that.